dspy/utils/usage_tracker.py

```diff
@@ -43,6 +43,8 @@ def _merge_usage_entries(self, usage_entry1, usage_entry2) -> dict[str, dict[str
                 else:
                     result[k] = result[k] or 0
                     result[k] += v if v else 0
+            else:
+                result[k] = dict(v) if isinstance(v, dict) else v
```

Comment:
probably use the following code to reduce nesting:

```
        result = dict(usage_entry2)
        for k, v in usage_entry1.items():
            current_v = result.get(k)
            if isinstance(v, dict):
                result[k] = self._merge_usage_entries(current_v, v)
            else:
                result[k] = current_v or 0
                result[k] += v if v else 0
        return result
```

Reply:
Nice edit